tests/selenium/test_basics.py

```diff
@@ -36,6 +36,9 @@ def test_perfherder_main(initial_data, live_server, selenium):
     This tests that the basic graphs view load and we can click the add tests button
     '''
     selenium.get(live_server.url + '/perf.html')
+    WebDriverWait(selenium, 10).until(
+        EC.presence_of_element_located((By.CSS_SELECTOR, '#graph canvas'))
```

Comment:
I'm probably misunderstanding how selenium/webdriver works (I know virtually nothing about them), but why is this wait required when there is already a wait below prior to the click? Is this just best practice (eg waiting on a page's outer components, so that if this test has additional cases added, it's still waiting on the correct thing?) or is this working around a bug that is causing the intermittent? If the latter, could you add a comment? :-)

I don't suppose you could

Reply:
It seems that waiting for the element to be displayed in this case is not always sufficient. This is potentially a bug in Marionette, but waiting for this additional element to be present works around it for me. I'll add a comment for sure.